modules/base/shaders/pointcloud/pointcloud_fs.glsl

```diff
@@ -85,21 +92,52 @@ Fragment getFragment() {
   }
 
   // Moving the origin to the center and calculating the length
-  float lengthFromCenter = length((texCoord - vec2(0.5)) * 2.0);
-  if (!hasSpriteTexture && (lengthFromCenter > 1.0)) {
+  vec2 centeredTexCoords = (texCoord - vec2(0.5)) * 2.0;
+  float lengthFromCenter = length(centeredTexCoords);
+
+  bool shouldBeRound = (!hasSpriteTexture && !enableOutline) ||
+    (enableOutline && outlineStyle == OutlineStyleRound);
+
+  if (shouldBeRound && (lengthFromCenter > 1.0)) {
     discard;
   }
 
   vec4 fullColor = vec4(color, 1.0);
+  vec4 cmapColor = vec4(1.0);
   if (useColorMap) {
-    fullColor = sampleColorMap(gs_colorParameter);
+    cmapColor = sampleColorMap(gs_colorParameter);
+    if (!useCmapOutline) {
```

Comment:
This also feels reversed, shouldn't the color map be used when the property is enabled?

Reply:
I understand that this can seem confusing. However, the color map color should only be the base color for the point if it isn't applied on the border/outline.  So this is correct